include/iterlib/MergeIterator.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2015 - present, Facebook
+
+#pragma once
+
+#include "iterlib/OrIterator.h"
+
+namespace iterlib {
+
+class MergeIterator : public UnionIterator {
```

Comment:
Probably a good idea to leave some comments on how this iterator should be implemented.